No matching member function for call to 'wait'

Just to mention #3094 introduced another compile error, which isn't fixed by #3100.

```
/media/sf_pcl/gpu/kinfu_large_scale/tools/record_maps_rgb.cpp:175:19: error: no matching member function for call to 'wait'
      buff_empty_.wait (buff_lock);
      ~~~~~~~~~~~~^~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/condition_variable:92:5: note: candidate function not viable: no known conversion from 'std::lock_guard<std::mutex>' to 'unique_lock<std::mutex> &' for 1st argument
    wait(unique_lock<mutex>& __lock) noexcept;
    ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/condition_variable:96:7: note: candidate function template not viable: requires 2 arguments, but 1 was provided
      wait(unique_lock<mutex>& __lock, _Predicate __p)
      ^
1 error generated.

```

@SergioRAgostinho: I believe this issue is for you, because you have more experience how to correct replace boost threading code ;).

--- a/gpu/kinfu_large_scale/tools/record_maps_rgb.cpp
+++ b/gpu/kinfu_large_scale/tools/record_maps_rgb.cpp
@@ -163,7 +163,7 @@ MapsBuffer::getFront(bool print)
 {
   boost::shared_ptr< const MapsBuffer::MapsRgb > depth_rgb;
   {
-    std::lock_guard<std::mutex> buff_lock (bmutex_);
+    std::unique_lock<std::mutex> buff_lock (bmutex_);
     while (buffer_.empty ())
     {
       if (is_done)

